test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -2756,6 +2756,171 @@ TEST_P(HttpFilterTestParam, NoCluster) {
+  InSequence s;
+
+  envoy::extensions::filters::http::ext_authz::v3::ExtAuthzPerRoute settings;
+  FilterConfigPerRoute auth_per_route(settings);
+
+  ON_CALL(*decoder_filter_callbacks_.route_, mostSpecificPerFilterConfig(_))
+      .WillByDefault(Return(&auth_per_route));
+
+  initialize(R"EOF(
+  transport_api_version: V3
+  grpc_service:
+    envoy_grpc:
+      cluster_name: "ext_authz_server"
+  failure_mode_allow: false
+  with_request_body:
+    max_request_bytes: 1
+    allow_partial_message: false
+  )EOF");
+
+  // Set allow_partial_message to true and max_request_bytes to 10 on the per-route filter.
+  envoy::extensions::filters::http::ext_authz::v3::BufferSettings buffer_settings;
+  buffer_settings.set_max_request_bytes(5);        // Set the max_request_bytes value
+  buffer_settings.set_allow_partial_message(true); // Set the allow_partial_message value
+  // Set the per-route filter config.
+  envoy::extensions::filters::http::ext_authz::v3::CheckSettings check_settings;
+  check_settings.mutable_with_request_body()->CopyFrom(buffer_settings);
+  settings.mutable_check_settings()->CopyFrom(check_settings);
+
+  // Initialize the route's per filter config.
+  auth_per_route = FilterConfigPerRoute(settings);
+
+  ON_CALL(decoder_filter_callbacks_, connection())
+      .WillByDefault(Return(OptRef<const Network::Connection>{connection_}));
+  ON_CALL(decoder_filter_callbacks_, decodingBuffer()).WillByDefault(Return(&data_));
+  ON_CALL(decoder_filter_callbacks_, addDecodedData(_, _))
+      .WillByDefault(Invoke([&](Buffer::Instance& data, bool) { data_.add(data); }));
+  EXPECT_CALL(decoder_filter_callbacks_, setDecoderBufferLimit(_)).Times(0);
+  connection_.stream_info_.downstream_connection_info_provider_->setRemoteAddress(addr_);
+  connection_.stream_info_.downstream_connection_info_provider_->setLocalAddress(addr_);
+  EXPECT_CALL(*client_, check(_, _, _, _));
+
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_->decodeHeaders(request_headers_, false));
+
+  Buffer::OwnedImpl buffer1("foo");
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationAndBuffer, filter_->decodeData(buffer1, false));
+  data_.add(buffer1.toString());
+
+  Buffer::OwnedImpl buffer2("bar");
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationAndBuffer, filter_->decodeData(buffer2, false));
+  data_.add(buffer2.toString());
+
+  Buffer::OwnedImpl buffer3("barfoo");
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationAndWatermark, filter_->decodeData(buffer3, true));
+  data_.add(buffer3.toString());
+
+  Buffer::OwnedImpl buffer4("more data after watermark is set is possible");
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationAndWatermark, filter_->decodeData(buffer4, true));
+
+  EXPECT_EQ(Http::FilterTrailersStatus::StopIteration, filter_->decodeTrailers(request_trailers_));
+}
+
+// Checks that the per-route filter can override the check_settings set on the main filter.
+TEST_F(HttpFilterTest, PerRouteCheckSettingsOverrideWorks) {
```

Comment:
This test is identical to the previous test except for the name

Reply:
Sorry, the previous test was meant to be slightly different where we don't initially specify the `with_request_body` on the main filter. I fixed it.